File: Scripts/552.cpp
class Solution {
public:
    int checkRecord(int n) {
      int MOD = 1e9 + 7;
      if(n == 1) return 3;
      long long all = 0;
      long long al = 1;
      long long ap = 1;
      long long aa = 2;
      long long nall = 1;
      long long nal = 1;
      long long nap = 2;
      
      for(int i = 3; i <= n; i++) {
        long long ALL = 0, AL = 0, AP = 0, AA = 0, NALL = 0, NAL = 0, NAP = 0;
        AP += all;
        AP += aa;
        AL += aa;
        ALL += al;
        AP += al;
        AP += ap;
        AL += ap;
        
        NAP += nall;
        AA += nall;
        NAP += nal;
        AA += nal;
        NALL += nal;
        NAP += nap;
        AA += nap;
        NAL += nap;
        
        all = ALL;
        al = AL;
        ap = AP;
        aa = AA;
        nall = NALL;
        nal  = NAL;
        nap = NAP;
        
        all %= MOD;
        al %= MOD;
        ap %= MOD;
        aa %= MOD;
        nall %= MOD;
        nal  %= MOD;
        nap %= MOD;
      }
      
      return (all + al + ap + aa + nall + nal + nap) % MOD;
    }
};

File: Scripts/552.go
func checkRecord(n int) int {
	mod := 1e9 + 7
	var all int64 = 0
	var al int64 = 1
	var ap int64 = 1
	var aa int64 = 2
	var nall int64 = 1
	var nal int64 = 1
	var nap int64 = 2

	for (let i = 1; i <= n; i++) {
		var ALL int64 = 0
		var AL int64 = 0
		var AP int64 = 0
		var AA int64 = 0
		var NALL int64 = 0
		var NAL int64 = 0
		var NAP int64 = 0
		AP += all + aa + al + ap
		AL += aa + ap
		ALL += al

		NAP += nall + nal + nap
		AA += nall + nal + nap
		NALL += nal + 
	}
	return int(all)
}